src/main/java/com/google/sps/utils/UserUtils.java

```diff
@@ -67,6 +66,32 @@ public static Entity getEntityFromDatastore(String entityName, String entityProp
     return entity;
   }
 
+     /**
+    * This function takes in a list of cookies and matches a sessionID cookie
+    * with a specific name/value pair to a user entity with the same name/value
+    * pair as one of its properties. This user entity is then returned. 
+    * 
+    * <p>
+    * 
+    * @param  cookies    an array of cookies (usually all cookies on front end)
+    * @param  datastore  the database where entities are stored
+    * @return            a single entity that has the cookie name/value pair as a property
+    */
+ 
+  public static Entity getUserFromCookie(Cookie cookies[], DatastoreService datastore){
+      if(datastore == null){
+          log.severe("Error in function getUserFromCookie(): Datastore passed in was null");
+          return null;
+      }
+    String sessionIDName = "SessionID";
+    Cookie cookie = CookieUtils.getCookieFromName(cookies, sessionIDName); //returns null if it doesn't exist
+    if(cookie == null){ 
+        log.severe("Error in function getUserFromCookie(): Cookie with name: " + sessionIDName + " not found.");
```

Comment:
I'd use `log.info` rather than `log.severe` for this. It's normal and expected for users sometimes to not be logged in, or even to have an invalid cookie.